src/modules/job-manager/plugins/update-duration.c

```diff
@@ -0,0 +1,90 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* update-duration.c - allow updates of attributes.system.duration for jobs
+ *
+ * This plugin implements a 'job.update.attributes.system.duration'
+ * callback to enable duration updates for pending jobs.
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <stdio.h>
+#include <jansson.h>
+#include <flux/core.h>
+#include <flux/jobtap.h>
+
+/*
+ *  Allow instance owner to update duration to any value, even if it
+ *  exceeds a configured duration limit. By default, this is true, to
+ *  disable this behavior, reload the `.update-duration` plugin with
+ *  owner-allow-any=0.
+ */
+static int owner_allow_any = 1;
+
+static int duration_update_cb (flux_plugin_t *p,
+                               const char *topic,
+                               flux_plugin_arg_t *args,
+                               void *arg)
+{
+    flux_jobid_t id;
```

Comment:
i don't think id is used in this function for anything

Reply:
Oh, yeah, good catch. Whatever was using the jobid got eventually moved out of that callback, so no longer necessary.